File: J_se_les12/HW/Core/Logic/Save/SaveV1.java
package Java.J_se_les12.HW.Core.Logic.Save;

import java.io.FileWriter;
import java.io.IOException;

import Java.J_se_les12.HW.Core.Contact.Contact;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public class SaveV1 extends SaveBase {

    public SaveV1(Phonebook phonebook) {
        super(phonebook);
    }

    @Override
    public void save(String type, String path) {
        try (FileWriter writer = new FileWriter(path, false)) {
            if (type.equals("txt")) {
                for (int i = 0; i < this.phonebook.sizeOfBook(); i++) {
                    Contact contact = this.phonebook.getContact(i);
                    writer.append(String.format("%s %s %s %d %s\n", contact.getfName(), contact.getlName(),
                            contact.getOrganisation(), contact.getPhoneNumber(), contact.getNote()));
                }
            } else if (type.equals("csv")) {
                for (int i = 0; i < this.phonebook.sizeOfBook(); i++) {
                    Contact contact = this.phonebook.getContact(i);
                    writer.append(String.format("%s;%s;\"%s\";%d;\"%s\"\n", contact.getfName(), contact.getlName(),
                            contact.getOrganisation(), contact.getPhoneNumber(), contact.getNote()));
                }
            }
            writer.flush();
            writer.close();
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }

    }

}

File: J_se_les13/HW/Core/Logic/Save/SaveV1.cs
using HW.Core.Phonebook;
using HW.Core.Contact;
namespace HW.Core.Logic.Save
{
    public class SaveV1 : SaveBase
    {
        public SaveV1(PhonebookV1 phonebook)
        : base(phonebook)
        {
        }

        public override void Save(string type, string path)
        {
            using (StreamWriter writer = new StreamWriter(path, false))
            {
                if (type == "txt")
                {
                    for (int i = 0; i < Phonebook.SizeOfBook(); i++)
                    {
                        ContactV1 contact = Phonebook.GetContact(i);
                        string text = $"{contact.GetfName()} {contact.GetlName()} {contact.GetOrganisation()} {contact.GetPhoneNumber()} {contact.GetNote()}";
                        writer.WriteLine(text);
                    }
                }
                else if (type == "csv")
                {
                    for (int i = 0; i < Phonebook.SizeOfBook(); i++)
                    {
                        ContactV1 contact = Phonebook.GetContact(i);
                        string text = $"{contact.GetfName()};{contact.GetlName()};\"{contact.GetOrganisation()}\";{contact.GetPhoneNumber()};\"{contact.GetNote()}\"";
                        writer.WriteLine(text);
                    }
                }
                writer.Flush();
                writer.Close();

            }
        }


    }
}